pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -304,6 +313,28 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req MetricsQueryReques
 		}
 	}
 
+	samplesProcessed := queryStats.LoadSamplesProcessedPerStep()
+	if len(samplesProcessed) > 0 {
+		// TODO: reuse stats.StepStat proto in Extent, instead of defining duplicate
```

Comment:
are you still doing this later?

Reply:
Yep, it seems Marco found the solution, but I better do it after I merge this one.